src/coreclr/vm/method.hpp

```diff
@@ -602,7 +689,8 @@ class MethodDesc
     {
         LIMITED_METHOD_DAC_CONTRACT;
         return mcFCall == GetClassification()
-            || mcArray == GetClassification();
+            || mcArray == GetClassification()
+            || IsAsyncThunkMethod();
```

Comment:
This is confusing method. It returns true for some runtime supplied methods and not others (e.g. it returns false for unboxing stubs). Is it a good idea to keep overloading it even more?

Reply:
`IsRuntimeSupplied()` felt like a meaningful thing at the time. I will use the same pattern as unboxing stubs then. 
This is not the only place where we see similarities with unboxing stubs.

Reply:
I've reverted `IsRuntimeSupplied()` to where it was.